pkg/apis/networking/types.go

```diff
@@ -699,3 +700,58 @@ type ClusterCIDRList struct {
 	// items is the list of ClusterCIDRs.
 	Items []ClusterCIDR
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPAddress represents an IP used in a Kubernetes cluster, commonly associated to a Service ClusterIP.
+// An IP address can be represented in different formats, to guarantee the uniqueness of the IP,
+// the name of the object is the IP address in canonical format, four decimal digits separated
+// by dots suppressing leading zeros for IPv4 and the representation defined by RFC 5952 for IPv6.
+// Valid: 192.168.1.5 or 2001:db8::1
+// Invalid: 10.01.2.3 or 2001:db8:0:0:0::1
+type IPAddress struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	// +optional
+	Spec IPAddressSpec
+}
+
+// IPAddressSpec describe the attributes in an IP Address,
+type IPAddressSpec struct {
+	// ParentRef references the resources that an IPAddress wants to be attached to.
```

Comment:
is it `IPAddress` wants to attach to it? or the other way around?

Reply:
is the IPAddress parent, we need this to define the relation Service.ClusterIP <--> IPAddress.
We tried to use OwnerReferences on the Service object, the goal was that destroying the parent , it  cascaded destroying the childs ... butds there is a limitation that namespaced objects can not reference non-namespaced objects

Reply:
Then it is 1:1 relationship. a better doc would be something like:

```
ParentRef references the resource that an IPAddress is attached to.
```

Reply:
got it